processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -1086,18 +1091,72 @@ private Column deserializeColumn(ObjectMapper mapper, ByteBuffer byteBuffer) thr
     }
   }
 
+  public static String makeDimFileName(String dimension)
+  {
+    return String.format("dim_%s.drd", sanitizeFileName(dimension));
+  }
+
   public static File makeDimFile(File dir, String dimension)
   {
-    return new File(dir, String.format("dim_%s.drd", dimension));
+    return new File(dir, makeDimFileName(dimension));
+  }
+
+  public static String makeTimeFileName(ByteOrder order)
+  {
+    return String.format("time_%s.drd", order);
   }
 
   public static File makeTimeFile(File dir, ByteOrder order)
   {
-    return new File(dir, String.format("time_%s.drd", order));
+    return new File(dir, makeTimeFileName(order));
+  }
+
+  public static String makeMetricFileName(String metricName, ByteOrder order)
+  {
+    return String.format("met_%s_%s.drd", sanitizeFileName(metricName), order);
   }
 
   public static File makeMetricFile(File dir, String metricName, ByteOrder order)
   {
-    return new File(dir, String.format("met_%s_%s.drd", metricName, order));
+    return new File(dir, makeMetricFileName(metricName, order));
+  }
+
+  public static String sanitizeFileName(String fileName)
+  {
+    String encoding = System.getProperty("file.encoding", "UTF-8");
+
+    final StringBuilder sb = new StringBuilder();
+    final String sha1 = DigestUtils.sha1Hex(fileName).substring(0, 6);
+    String fnameBase = fileName.replace(File.separator, "_");
+
+    try {
+      fnameBase = new URI(String.format("file:/%s", fnameBase)).toASCIIString().substring("file:/".length());
+    }
+    catch (URISyntaxException e) {
+      log.debug(e, "Unable to encode base name [%s]. Trying as hex of UTF-8.", fnameBase);
+      fnameBase = javax.xml.bind.DatatypeConverter.printHexBinary(StringUtils.toUtf8(fnameBase));
+    }
+
+    // For REALLY long names we truncate
+    byte[] fnameBytes;
+    do {
+      try {
+        fnameBytes = fnameBase.getBytes(encoding);
+      }
+      catch (UnsupportedEncodingException e) {
+        log.debug(e, "Error coding [%s] in %s, defaulting to UTF-8", fnameBase, encoding);
+        fnameBytes = StringUtils.toUtf8(fnameBase);
+      }
+      if (fnameBytes.length > 100) {
+        if (fnameBase.length() > 1) {
+          fnameBase = fnameBase.substring(0, fnameBase.length() / 2);
+        } else {
+          // length of 1 but byte encoding is > 100... wtf kind of encoding allows that?
+          fnameBase = "";
+        }
+      }
+    } while (fnameBytes.length > 100 && !fnameBase.isEmpty());
+
+    return sb.append(fnameBase).append('.').append(sha1).toString();
```

Comment:
you can just use + for readability, the StringBuilder is a bit overkill for that. Internally the JVM will probably use a StringBuilder anyway.